### PR TITLE
QUARKUS-2991 Support for MicroProfile 6

### PR DESCRIPTION
**Pull request by jsmrcka**
### Links

JIRA: https://issues.redhat.com/browse/QUARKUS-2991

Quarkus documentation:
- https://quarkus.io/version/main/guides/opentelemetry
- https://quarkus.io/version/main/guides/openapi-swaggerui
- https://quarkus.io/version/main/guides/security-jwt
- https://quarkus.io/version/main/guides/smallrye-metrics

MicroProfile documentation:
- https://microprofile.io/2023/01/10/microprofile-6-0-release
- https://microprofile.io/2021/12/07/microprofile-5-0-release

### Reminder for considerable topics

 - [ ] Make sure you have considered the following areas when preparing the test plan:

   - Logging
   - Tracing
   - Metrics
   - Security
   - OpenAPI
   - Data sources
   - Frontend
   - Qute

### CODE REVIEW

File: QUARKUS-2991.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
+# QUARKUS-2991 Support for MicroProfile 6
+
+JIRA link: https://issues.redhat.com/browse/QUARKUS-2991
+
+Quarkus documentation:
+- https://quarkus.io/version/main/guides/opentelemetry
+- https://quarkus.io/version/main/guides/openapi-swaggerui
+- https://quarkus.io/version/main/guides/security-jwt
+- https://quarkus.io/version/main/guides/smallrye-metrics
+
```
Comment (rsvoboda) [inline QUARKUS-2991.md:10]: Add pointers to MP stuff

 - https://microprofile.io/2023/01/10/microprofile-6-0-release/#
 - https://microprofile.io/2021/12/07/microprofile-5-0-release/#

File: QUARKUS-2991.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+- Config 3.0
+- Fault Tolerance 4.0
+- JWT Authentication  2.1 - updated in MP 6
+- Health 4.0
+- Jakarta EE 10 Core Profile - updated in MP 6
+
+RHBQ 3.x should also support:
+- Reactive Messaging 3.0 - updated in MP 6
+- Reactive Streams Operators 3.0 - updated in MP 6
+- Context Propagation 1.3
```
Comment (rsvoboda) [inline QUARKUS-2991.md:29]: Updated in MP 5 which Quarkus intentionally skipped and stayed on MP 4.1 specs

File: QUARKUS-2991.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+#### Priority
+
+Telemetry 1.0
+- Currently, the TCK is inactive in Quarkus `main` due to an API incompatibility in the latest MP release.
+  An update of the TCK in MP repository is beyond Quarkus' scope.
+- AIs:
+  - Asses the scope of the test coverage in the Quarkus integration tests.
+  - Asses the existing tests in QE TS.
+  - Develop new integration test scenario in QE TS.
+
```
Comment (rsvoboda) [inline QUARKUS-2991.md:101]: Is Reactive Messaging test coverage sufficient or it should be also considered for TD addition?

**Review by rsvoboda — CHANGES_REQUESTED** [review]

File: QUARKUS-2991.md (hunk 2 above)
Comment (jsmrcka) [reply QUARKUS-2991.md:33]: Mentioned further in the TP.

File: QUARKUS-2991.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+- AI: Check the decline in number of tests in `smallrye-context-propagation` module of integration tests.
+
+Jakarta EE 10 Core Profile
+- New updates, no tests found.
+- AI: TCK or TD is needed.
+
+Reactive Streams Operators 3.0
+- New updates, no tests found.
+- AI: TCK or TD is needed.
+
```
[the remark this answers is not in the document]
Comment (jsmrcka) [reply QUARKUS-2991.md:128]: Added OpenTracing.

GraphQL - no functional changes in MP 5/6, we have existing TCKs, Quarkus integration tests and QE TS integration tests, so I didn't mention it.

[the remark this answers is not in the document]
Comment (rsvoboda) [reply QUARKUS-2991.md:128]: You have analyzed it, it should be tracked

**Review by rsvoboda — APPROVED** [review]

**Pull request merged**